client/src/components/profile/components/portfolio.tsx

```diff
@@ -76,18 +82,29 @@ class PortfolioSettings extends Component<PortfolioProps, PortfolioState> {
     (e: React.ChangeEvent<HTMLInputElement>) => {
       e.preventDefault();
       const userInput = e.target.value.slice();
-      return this.setState(state => {
-        const { portfolio: currentPortfolio } = state;
-        const mutablePortfolio = currentPortfolio.slice(0);
-        const index = findIndex(currentPortfolio, p => p.id === id);
-
-        mutablePortfolio[index] = {
-          ...mutablePortfolio[index],
-          [key]: userInput
-        };
-
-        return { portfolio: mutablePortfolio };
-      });
+      this.setState(
+        state => {
+          const { portfolio: currentPortfolio } = state;
+          const mutablePortfolio = currentPortfolio.slice(0);
+          const index = findIndex(currentPortfolio, p => p.id === id);
+
+          mutablePortfolio[index] = {
+            ...mutablePortfolio[index],
+            [key]: userInput
+          };
+
+          return { portfolio: mutablePortfolio };
+        },
+        () => {
+          if (key === 'image' && userInput) {
+            void this.validateImageLoad(userInput).then(imageValidation => {
+              this.setState({ isImageValid: imageValidation });
+            });
+          } else if (key === 'image' && !userInput) {
+            this.setState({ isImageValid: { state: 'success', message: '' } });
+          }
+        }
+      );
```

Comment:
It makes sure the state it the latest one. Which is a pain with class components.

Reply:
I don't believe that's how it works. The point of the callback is simply to deal with the fact that `this.state` does not change immediately after `this.setState` is called. e.g.

```
// this.state starts as { thing: false }
this.setState({ thing: true })
if (this.state.thing) {
 // this branch isn't reached because the state has not yet changed.
}
```

Since we don't care about the current state and the only use for the callback is when we _do_ care, this is misleading.